dtschema/schemas/options/u-boot.yaml

```diff
@@ -71,6 +71,21 @@ properties:
       2: use simplified command line (e.g. avoid hush)
       3... reserved
 
+  bootscr-address:
+    $ref: types.yaml#/definitions/uint64
+    default: 0x0
+    description:
+      This property holds the full address of the boot script file. This helps in
+      making automated flows easier by fetching address directly from DT.
```

Comment:
Can you add a bit more detail, or point to docs in the U-Boot tree? How does the script end up in memory at that address? In what circumstances is it actually run?

Reply:
In our implementation scriptaddr variable is setup based on DT when variables are not saved by SOC. 
And scriptaddr is mandatory variable for distro boot and store address where boot.scr script is loaded prior execution.
https://github.com/u-boot/u-boot/blob/master/doc/develop/distro.rst?plain=1#L291

It shouldn't be a problem to extend documentation and point to exact variable name and it's usage.

Reply:
OK I see. We are moving away from distroboot and to standard boot.

I agree it makes sense to put these variables in the DT rather than the environment, at least as a default.

In that case how about having a subnode in options, like:

options {
   bootstd {
      compatible = "u-boot,bootstd";
      script-addr = /bits 64/ <...>;
      script-ram-offset =...
   }

as I suspect we will end up adding all the other env vars in here too.

Reply:
Why this should be different compatible if we have u-boot,config already? 
For example bootdelay-sec or bootcmd are also related to boot process. Does it mean that we should move it here too? 

Pretty much all of these properties should be moved here. 
https://github.com/u-boot/u-boot/blob/master/doc/device-tree-bindings/config.txt